src/odemis/driver/zeiss.py

```diff
@@ -0,0 +1,983 @@
+
+    @isasync
+    def moveRel(self, shift):
+        """
+        shift (dict): shift in m
+        """
+        if not shift:
+            return model.InstantaneousFuture()
+        self._checkMoveRel(shift)
+        foc = self.parent.GetFocus()  #  mm
+        foc += shift["z"] * 1e3
+
+        f = model.CancellableFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, f, foc)
+        return f
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        pos (dict): pos in m
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+
+        foc = pos["z"] * 1e3
+        f = model.CancellableFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, f, foc)
+        return f
+
+    def stop(self, axes=None):
+        """
+        Stop the last command
+        """
+        # Empty the queue (and already stop the stage if a future is running)
+        self._executor.cancel()
+        logging.debug("Stopping all axes: %s", ", ".join(self.axes))
+
+        try:
+            self._updatePosition()
+        except Exception:
+            logging.exception("Unexpected failure when updating position")
+
+
+class RemconSimulator(object):
+    """
+    Simulates a Keithley 6485
+    Same interface as the serial port
+    """
+
+    def __init__(self, timeout=1, *args, **kwargs):
+        self.timeout = timeout
+        self._output_buf = ""  # what the commands sends back to the "host computer"
+        self._input_buf = ""  # what we receive from the "host computer"
+
+        self._speed = 10  # um/s
+        self._hfw_nomag = 1  # m
+        self.eol = '\r\n'
+
+        # Initialize parameters
+        self.magnification = 10
+        self.horizontalFoV = 5
+        self.pixelSize = 5
+        self.blanker = 0
+        self.external = 0
+        self.posX = 20
+        self.posY = 20
+        self.posZ = 5
+        self.focus = 0
+        self.eht = 0
+        self.pc = 1e-6
+        self.dur = 0
+        # Prepare moving thread
+        self.target_posX, self.target_posY, self.target_posZ = 0, 0, 0
+        self._end_move = 0
+        self._stage_stop = threading.Event()
+        self._is_moving = False
+        
+    def _run_move(self):
+        orig_posX, orig_posY, orig_posZ = self.posX, self.posY, self.posZ
+        while time.time() < self._end_move:
+            if self._stage_stop.wait(0.01):
+                logging.debug("SIM: Aborting move")
+                break
+            total_dist = abs(numpy.subtract((self.target_posX, self.target_posY, self.target_posZ),
+                             (self.posX, self.posY, self.posZ)))
+            traveled_dist = total_dist * (time.time() - self._start_move) \
+                            / (self._end_move - self._start_move)
+            self.posX, self.posY, self.posZ = (orig_posX, orig_posY, orig_posZ) \
+                                                + traveled_dist
+        if not self._stage_stop:
```

Comment:
if you replace line 855 with         
if not self._stage_stop.wait(0.01):

my simulator passes all tests